src/Kiota.Builder/Extensions/StringExtensions.cs

```diff
@@ -150,7 +150,14 @@ public static string CleanupSymbolName(this string? original)
     {
         if (string.IsNullOrEmpty(original)) return string.Empty;
 
-        var result = propertyCleanupRegex.Replace(original,
+        string result = (String)original.Clone();
+        foreach (var norm in NormalizedTrailingSymbols)
+        {
+            if (result.StartsWith(norm.Key, StringComparison.OrdinalIgnoreCase))
+                result = NormalizeTrailingSymbol(result, norm.Value.Item1, norm.Value.Item2);
+        }
+
+        result = propertyCleanupRegex.Replace(result,
```

Comment:
this is very specific to the github reactions case. I'd prefer to expand the numbers scenario to signs instead and include plus and minus in there.

Reply:
I had the same reaction at first, but, looking at the implementation, the substitution of numbers is pretty specific, with explicit parsing(i.e. `int.TryParse`) etc.
Also, the application of the fix happens in two different phases (before and after the cleaning respectively).

This implementation is instead generic toward symbols and I'm providing a specific implementation for plus and minus as a first iteration.

Reply:
I started trying to do full replacements, but, `StartsWith` is much less expensive than `Contains`, and mid-string characters are better to be preserved, i.e.:
`kebab-case` would have been translated to `kebab_minus_case` which, I think, is a regression in terms of the idiomaticity of the produced code.

Reply:
right, but 1+1 will still translation to an invalid symbol of One+One, correct?

Reply:
Correct, I don't have a "one size fits all" solution for this :-( if you have any proposal feel free to share!
As an alternative we can open a separate issue to discuss and eventually iterate over this point, it's a not a huge amount of LOCs and, I believe, this proposal slightly improve over the current situation (despite not being a perfect solution).

Reply:
the costliest operation in all that is probably parsing the regex for each symbol. Moving from starts with + regex to just string replace (which would no-op when the symbol is not present) would probably be much faster.
The only reason why I had a regex for numbers is because numbers are only invalid at the beginning of a code symbol, and I needed the numeric value to express the replacement correctly.

Reply:
Yep, this makes sense now, I'll fix it tomorrow :+1:

Reply:
@baywet I ended up manually writing the logic for each case, I'm not sure if I like this implementation tbh, but I'm afraid that this detail is keeping this PR on hold (and it already contains other valuable fixes).

What do you think about separating this discussion into a follow-up issue? (Feel free to assign it to me)